azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/AzureOperationAspect.java

```diff
@@ -28,38 +28,31 @@ public void operation() {
 
     @Before("operation()")
     public void beforeEnter(JoinPoint point) {
-        enterOperation(point);
+        final AzureOperationRef operation = toOperationRef(point);
+        AzureTelemeter.beforeEnter(operation);
+        AzureTaskContext.current().pushOperation(operation);
     }
 
     @AfterReturning("operation()")
     public void afterReturning(JoinPoint point) {
-        exitOperation(point);
+        final AzureOperationRef current = toOperationRef(point);
+        final AzureOperationRef operation = (AzureOperationRef) AzureTaskContext.current().popOperation();
+        AzureTelemeter.afterExit(operation);
+        assert Objects.equals(current, operation) : String.format("popped operation[%s] is not the exiting operation[%s]", current, operation);
     }
 
     @AfterThrowing(pointcut = "operation()", throwing = "e")
     public void afterThrowing(JoinPoint point, Throwable e) throws Throwable {
-        final AzureOperationRef operation = exitOperation(point);
+        final AzureOperationRef current = toOperationRef(point);
+        final AzureOperationRef operation = (AzureOperationRef) AzureTaskContext.current().popOperation();
+        AzureTelemeter.onError(operation, e);
```

Comment:
Do we need to remove other `AzureTelemeter.onError` call, like codes in `AzureTaskContext` in case duplicated?

Reply:
no, they are not duplicated, `onError` in `AzureTaskContext` is only for `AzureTask`. this is for `AzureOperation` only.